share/completions/p4.fish

```diff
@@ -0,0 +1,803 @@
+            echo b\t"Ignore whitespace-only changes (for instance, a tab replaced by eight spaces)"
+            echo w\t"Ignore whitespace altogether (for instance, deletion of tabs or other whitespace)"
+            echo l\t"Ignore differences in line-ending convention"
+    end
+end
+
+#########################################################
+
+function __fish_print_p4_parallel_options
+    set -l mode
+    if test -n "$argv"
+        set mode $argv[1]
+    end
+
+    # for now only looks that mode is set, later it will need to have a specific setting
+    echo 'threads='\t"sends files concurrently using N independent network connections"
+    echo 'batch='\t"specifies the number of files in a batch"
+    test "$mode"
+    or echo 'batchsize='\t"specifies the number of bytes in a batch"
+    echo 'min='\t"specifies the minimum number of files in a parallel sync"
+    test "$mode"
+    or echo 'minsize='\t"specifies the minimum number of bytes in a parallel sync"
+end
+
+#########################################################
+
+function __fish_print_p4_submit_options
+    echo 'submitunchanged'\t"Submit all open files (default behavior)"
+    echo 'submitunchanged+reopen'\t"Submit all open files + reopen to default changelist"
+    echo 'revertunchanged'\t"Revert unmodified files, submit all the rest"
+    echo 'revertunchanged+reopen'\t"Revert unmodified files, submit all the rest + reopen them in default changelist"
+    echo 'leaveunchanged'\t"Move unchanged files to default changelist, submit all the rest"
+    echo 'leaveunchanged+reopen'\t"Submit only modified files + reopen all (modified and unmodified) in default changelist"
+end
+
+#########################################################
+
+function __fish_print_p4_noretransfer_options
+    echo 1\t"server avoids re-transferring files that have already been archived after a failed submit operation"
+    echo 0\t"server re-transfers all files after a failed submit operation"
+end
+
+#########################################################
+
+function __fish_print_p4_integrate_output_options
+    echo b\t"outputs the base revision for the merge (if any)"
+    echo r\t"outputs the resolves that are being scheduled"
+end
+
+#########################################################
+
+function __fish_print_p4_integrate_resolve_options
+    echo b\t"schedules a branch resolve instead of branching the target files automatically"
+    echo d\t"schedules a delete resolve instead of deleting the target files automatically"
+    echo s\t"skips cherry-picked revisions that have already been integrated"
+end
+
+
+#########################################################
+
+function __fish_p4_in_subcommand
+    for i in (commandline -opc)
+        if contains -- $i (__fish_print_p4_commands_list)
+            return 0
+        end
+    end
+    return 1
+end
+
+#########################################################
+
+# Negating __fish_p4_in_subcommand in complete -n doesn't seem to work,
```

Comment:
`complete -c p4 -n 'not __fish_p4_in_subcommand'` doesn't work? You probably forgot the quotes.